tools/Tool.cc

```diff
@@ -51,6 +57,61 @@ static bool outputIsColor() {
     return sColor;
 }
 
+static bool tokenize(const char* input, deque<string> &args)
+{
+    string str(input);
```

Comment:
Turning the input into a C++ string doesn't seem worth it. The nice thing with parsing a C string is you can treat the trailing NUL byte as a special character, which simplifies parsing. Here you have to do some special checks to handle finishing the last argument, mismatched quotes, etc.